File: python/p023.py
from itertools import combinations_with_replacement
from utils import divisors


abundant = filter(lambda n: sum(divisors(n)) > n, range(2, 29000))

u = set(range(1, 29000))
for i in combinations_with_replacement(abundant, 2):
    u.discard(sum(i))

print sum(u)

File: erlang/p023.erl
-module(p023).

-export([start/0]).

f() ->
  A = [ N || N <- lists:seq(2, 29000), lists:sum(utils:divisors(N)) > N ],
  lists:sum(g(sets:from_list(lists:seq(1, 29000)), A)).

g(R, []) -> sets:to_list(R);
g(R, [H|T]) -> g(sets:subtract(R, sets:from_list([ H + X || X <- [H|T] ])), T).

start() ->
  io:format("~p ~n", [f()]).

